lost/src/main/java/com/mapzen/android/lost/internal/FusedLocationProviderApiImpl.java

```diff
@@ -30,95 +28,81 @@
-            callbacks.onConnectionSuspended();
-          }
-        }
-        connectState = ConnectState.IDLE;
+    @Override public void onDisconnect(LostApiClient client, boolean stopService,
+        boolean disconnectService) {
+      if (disconnectService) {
+        service.disconnect(client);
+      }
+      if (stopService) {
+        context.unbindService(serviceConnection);
+        Intent intent = new Intent(context, FusedLocationProviderService.class);
+        context.stopService(intent);
+        service = null;
       }
-      Log.d(TAG, "[onServiceDisconnected]");
     }
   };
 
-  Set<LostApiClient.ConnectionCallbacks> connectionCallbacks;
+  private final ServiceConnection serviceConnection = new ServiceConnection() {
+    @Override public void onServiceConnected(ComponentName name, IBinder binder) {
+      serviceConnectionManager.onServiceConnected(binder);
+    }
 
+    @Override public void onServiceDisconnected(ComponentName name) {
+      serviceConnectionManager.onServiceDisconnected();
+    }
+  };
+
+  public FusedLocationProviderApiImpl(FusedLocationServiceConnectionManager connectionManager) {
+    serviceConnectionManager = connectionManager;
+    serviceConnectionManager.setEventCallbacks(eventCallbacks);
+  }
 
   public boolean isConnecting() {
-    return connectState == ConnectState.CONNECTING;
+    return serviceConnectionManager.isConnecting();
   }
 
-  public FusedLocationProviderApiImpl() {
-    connectionCallbacks = new HashSet<>();
-    connectState = ConnectState.IDLE;
+  public void addConnectionCallbacks(LostApiClient.ConnectionCallbacks callbacks) {
+    serviceConnectionManager.addCallbacks(callbacks);
   }
 
   public void connect(Context context, LostApiClient.ConnectionCallbacks callbacks) {
-    if (connectState == ConnectState.IDLE) {
-      this.context = context;
-      connectState = ConnectState.CONNECTING;
-
-      Intent intent = new Intent(context, FusedLocationProviderService.class);
-      context.startService(intent);
-
-      intent = new Intent(context, FusedLocationProviderService.class);
-      context.bindService(intent, serviceConnection, Context.BIND_AUTO_CREATE);
-    }
-    if (callbacks != null) {
-      connectionCallbacks.add(callbacks);
-    }
+    serviceConnectionManager.connect(context, callbacks);
   }
 
   public void disconnect() {
     disconnect(null, true);
   }
 
   public void disconnect(LostApiClient client, boolean stopService) {
-    if (connectState != ConnectState.IDLE) {
-      if (connectState == ConnectState.CONNECTED) {
-        service.disconnect(client);
-      }
-      connectState = ConnectState.IDLE;
-      if (stopService) {
-        context.unbindService(serviceConnection);
-        Intent intent = new Intent(context, FusedLocationProviderService.class);
-        context.stopService(intent);
-        service = null;
-      }
-    }
+    serviceConnectionManager.disconnect(client, stopService);
```

Comment:
I don't have an answer to that- removing it as you're right, its only ever called with `true`